xapian-core/weight/boolweight.cc

```diff
@@ -86,4 +86,14 @@ BoolWeight::get_maxextra() const
     return 0;
 }
 
+BoolWeight *
+BoolWeight::create_from_parameters(const char * p) const
+{
+    if (*p == '\0')
+	return new Xapian::BoolWeight();
+    if (*p == ' ')
+	throw "No parameters are required for BoolWeight";
+    return new Xapian::BoolWeight();
```

Comment:
Currently `"1.7"` is a valid parameter string for `BoolWeight`, which seems wrong.  Any non-empty parameter string should probably throw an exception.

Also, the exception type should be `Xapian::InvalidArgumentError`, not `const char*`.

The same comments apply to several other weighting schemes below.